pydantic/main.py

```diff
@@ -428,8 +428,28 @@ def json(
             exclude_defaults=exclude_defaults,
             exclude_none=exclude_none,
         )
-        if self.__custom_root_type__:
-            data = data[ROOT_KEY]
+
+        def reparent_root(d: 'DictStrAny') -> 'DictStrAny':
```

Comment:
I think this can be done more easily with a change to `_get_value`, though I haven't tried myself.

Reply:
I tried to look at _get_value but it would spread the change also to dict() method and make some other tests failing (and might introduce new issues?).